src/coreComponents/physicsSolvers/multiphysics/MultiphasePoroelasticSolver.hpp

```diff
@@ -0,0 +1,149 @@
+
+  virtual void
+  implicitStepSetup( real64 const & time_n,
+                     real64 const & dt,
+                     DomainPartition & domain ) override;
+
+  virtual void
+  assembleSystem( real64 const time,
+                  real64 const dt,
+                  DomainPartition & domain,
+                  DofManager const & dofManager,
+                  CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                  arrayView1d< real64 > const & localRhs ) override;
+
+  virtual void
+  applyBoundaryConditions( real64 const time_n,
+                           real64 const dt,
+                           DomainPartition & domain,
+                           DofManager const & dofManager,
+                           CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                           arrayView1d< real64 > const & localRhs ) override;
+
+  virtual real64
+  calculateResidualNorm( DomainPartition const & domain,
+                         DofManager const & dofManager,
+                         arrayView1d< real64 const > const & localRhs ) override;
+
+  virtual void
+  solveSystem( DofManager const & dofManager,
+               ParallelMatrix & matrix,
+               ParallelVector & rhs,
+               ParallelVector & solution ) override;
+
+  virtual void
+  applySystemSolution( DofManager const & dofManager,
+                       arrayView1d< real64 const > const & localSolution,
+                       real64 const scalingFactor,
+                       DomainPartition & domain ) override;
+
+  virtual void
+  implicitStepComplete( real64 const & time_n,
+                        real64 const & dt,
+                        DomainPartition & domain ) override;
+
+  virtual void
+  resetStateToBeginningOfStep( DomainPartition & domain ) override;
+
+  virtual real64
+  solverStep( real64 const & time_n,
+              real64 const & dt,
+              int const cycleNumber,
+              DomainPartition & domain ) override;
+
+  enum class CouplingTypeOption : integer
+  {
+    FIM
+  };
+
+  struct viewKeyStruct : SolverBase::viewKeyStruct
+  {
+    constexpr static char const * couplingTypeOptionString() { return "couplingTypeOptionEnum"; }
+    constexpr static char const * couplingTypeOptionStringString() { return "couplingTypeOption"; }
+
+    constexpr static char const * solidSolverNameString() { return "solidSolverName"; }
+    constexpr static char const * fluidSolverNameString() { return "fluidSolverName"; }
+  };
+
+
+
+protected:
+
+  virtual void postProcessInput() override;
+
+  string m_solidSolverName;
+  string m_flowSolverName;
+
+  CouplingTypeOption m_couplingTypeOption;
+
+  // pointer to the flow sub-solver
+  CompositionalMultiphaseBase * m_flowSolver;
+
+  // pointer to the solid mechanics sub-solver
+  SolidMechanicsLagrangianFEM * m_solidSolver;
+
+};
+
+ENUM_STRINGS( MultiphasePoroelasticSolver::CouplingTypeOption, "FIM", "SIM_FixedStress" )
```

Comment:
This has an extra string that does not correspond to anything in enum....
Unfortunately there's no mechanism to enforce this list being consistent with enum definition.

Reply:
Done